### PR TITLE
CATKIT-25: refactor iris command interface to streamline and better use poppy infrastructure

### PR DESCRIPTION
**Pull request by mperrin**
Implements previous suggestions in https://jira.stsci.edu/browse/CATKIT-25; see 
- https://github.com/spacetelescope/instrument-interface-library/pull/71#discussion_r463586030
- https://github.com/spacetelescope/instrument-interface-library/pull/71#discussion_r465338237

In particular this includes: 

- [x]  Making `SegmentedAperture` subclass from `poppy.dms.HexSegmentedDeformableMirror`
- [x]  Making `SegmentedDmCommand` no longer a subclass of `SegmentedAperture`; instead it has an `aperture` member that is the `SegmentedAperture` instance. 
- [x] The `PoppySegmentedCommand` is renamed `PoppySegmentedDmCommand` and becomes a proper subclass of `SegmentedDmCommand`.

It also includes
- [x]  Renaming `SegmentedDmCommand`'s `wavelength` parameter and attribute to `display_wavelength`, to make it clearer that the wavelength is only used in displays and doesn't actually affect the command values at all. The wavelength parameter and attribute are removed from the `SegmentedAperture` class since they're not used there. 
- [x]  The wavelength parameter gets a default value of 640 nm, as suggested in https://jira.stsci.edu/browse/CATKIT-48. 
- [x]  Likewise add `display_` to the `instrument_fov` and `pixelscale` attributes too. The values for these are now read from the config file in the section name given by `dm_config_id`, removing any dependency on the "testbed" section. This is https://jira.stsci.edu/browse/CATKIT-49

Since this is an interface refactor I tagged lots of people to take a look for more visibility.  For HICAT, there's an associated PR to go along with this. I haven't looked yet at what JOST code would need to be updated after this, but some changes are almost certainly needed.

### CODE REVIEW

File: catkit/hardware/iris_ao/segmented_dm_command.py
Comment (ivalaginja) [inline catkit/hardware/iris_ao/segmented_dm_command.py]: This parameter is now also called `display_wavelength`

File: catkit/hardware/iris_ao/segmented_dm_command.py
Comment (ivalaginja) [inline catkit/hardware/iris_ao/segmented_dm_command.py]: It might be worth calling the param here `display_wavelength` as well, and add it to the docstring.

File: catkit/hardware/iris_ao/segmented_dm_command.py
Comment (ivalaginja) [inline catkit/hardware/iris_ao/segmented_dm_command.py:221]: [revision] Double-check that you now use the attributes of this attribute. I found one leftover in line 242 ([here](https://github.com/spacetelescope/catkit/blob/decfc1b319a1b1dd609c8ed2301e39fc86e1a3fd/catkit/hardware/iris_ao/segmented_dm_command.py#L242)).

File: catkit/hardware/iris_ao/segmented_dm_command.py
Comment (ivalaginja) [inline catkit/hardware/iris_ao/segmented_dm_command.py]: this can go

File: catkit/hardware/iris_ao/segmented_dm_command.py
Comment (ivalaginja) [inline catkit/hardware/iris_ao/segmented_dm_command.py]: You didn't change this, but I am now wondering what is meant with the `DisplayCommand` class...

File: catkit/hardware/iris_ao/segmented_dm_command.py
Comment (ivalaginja) [inline catkit/hardware/iris_ao/segmented_dm_command.py]: You changed the attribute name in the docstring but not the actual attribute itself.

**Review by ivalaginja — CHANGES_REQUESTED** [review]
Also on this side, only some minor comments.

File: catkit/hardware/iris_ao/segmented_dm_command.py
Comment (mperrin) [reply catkit/hardware/iris_ao/segmented_dm_command.py]: I just rewrote this whole comment string to be more up-to-date for this revised interface.

File: catkit/hardware/iris_ao/segmented_dm_command.py
Comment (mperrin) [reply catkit/hardware/iris_ao/segmented_dm_command.py]: No idea. I can't find any instance of `DisplayCommand` anywhere in all of catkit, so I deleted this remark.

**Comment by ivalaginja** [comment]
@mperrin would you mind rebasing before I review again?

**Review by ivalaginja — DISMISSED** [review]
Good by me, ping me again if you need after rebasing/tests passing.

**Review by ivalaginja — DISMISSED** [review]

**Comment by mperrin** [comment]
@jamienoss - since this PR revises the interface between catkit and hicat for the iris control, should I also increment the version number in this? And then increment the dependency in the HICAT-959 PR?

**Comment by jamienoss** [comment]
@mperrin I'll do the catkit one, but go ahead on the hicat side, but note that the tests will fail until this is merged, bumped, and tagged.

File: catkit/hardware/iris_ao/segmented_dm_command.py
Comment (kjbrooks) [reply catkit/hardware/iris_ao/segmented_dm_command.py]: `DisplayCommand` is just a ghost of old code and I forgot to update the docstring.

File: catkit/hardware/iris_ao/segmented_dm_command.py
Comment (kjbrooks) [inline catkit/hardware/iris_ao/segmented_dm_command.py]: (Disclaimer: I recognize that you aren't necessarily the author of the parts that I am commenting on but since this was updated it drew my attention to it) 

I am pretty uncomfortable with needing to include a wavelength when loading a command. I recommend removing it as an input parameter for `SegmentedDmCommand`, and only require it as a parameter for the `display` method.

File: catkit/hardware/iris_ao/segmented_dm_command.py
Comment (kjbrooks) [inline catkit/hardware/iris_ao/segmented_dm_command.py]: since the wavelength isn't used for the display in this class, it seems odd that it is called the `display_wavelenth`. You might want to think of a more general variable name.

**Review by kjbrooks — COMMENTED** [review]
I had just a couple comments with a pretty narrow scope. This looks good to me. 

Because this will impact JOST and will require changes there, I really appreciate that you requested and waited for my review before merging the changes!

**Comment by kjbrooks** [comment]
To make sure these are linked, this PR impacts https://github.com/spacetelescope/jost-package/pull/15

File: catkit/hardware/iris_ao/segmented_dm_command.py
Comment (mperrin) [reply catkit/hardware/iris_ao/segmented_dm_command.py]: > I am pretty uncomfortable with needing to include a wavelength when loading a command. I recommend removing it as an input parameter for SegmentedDmCommand, and only require it as a parameter for the display method.

Great - I'm all for that change too! I wasn't sure how much I should minimize changes to avoid complications for JOST, but I definitely agree it's better to remove any of the display stuff from when needing to load a command. I'll add that in here too. Thanks.

**Review by jamienoss — COMMENTED** [review]
I've tried to stay out of ``catkit.hardware.iris_ao.segmented_dm_command`` and since this PR doesn't touch the actual dm control code, I'm gonna defer to the other reviewers.

**Comment by kjbrooks** [comment]
Is there a plan to test the changes on hardware? If we can wait a little, I can test this on JOST later today.

**Comment by ivalaginja** [comment]
@kjbrooks I tested it on the spare and the simulator when Marshall asked for a review the first time around, and can do that again when it's ready.

**Comment by kjbrooks** [comment]
@ivalaginja sounds good, I would also like to test it on JOST, as long as everyone is willing to wait a few hours

**Comment by jamienoss** [comment]
Let's hold off on merging thus until https://github.com/spacetelescope/catkit/pull/149 is merged, please.

**Comment by ivalaginja** [comment]
@jamienoss no problem

**Comment by ivalaginja** [comment]
Tests on spare hardware and in sim work, and #149 is merged, so just waiting for an ok from @kjbrooks now.

**Comment by jamienoss** [comment]
@ivalaginja When this gets merged ping me so I can create a new release for the downstream PR to use, please.

**Review by ivalaginja — APPROVED** [review]

**Review by kjbrooks — APPROVED** [review]
I realized that I had forgotten that in order to use the latest IrisAO controls, I need @jamienoss to rebuild the c code on the JOST machine which he can't do until I finish the JOST PR I am currently working on (that feeling when you depend on yourself to complete a task that is needed for a different project that you are working on...). Because of this, and the urgency on this PR, I am approving it and will open an issue for any problems I might have when I can eventually test this. Thanks!

**Comment by ivalaginja** [comment]
@jamienoss time for you to step in

**Comment by jamienoss** [comment]
Downstream hicat tests failing due to dependency on https://github.com/spacetelescope/hicat-package/pull/410

**Comment by jamienoss** [comment]
@kjbrooks I'll take another look but I'm not sure if it's true that it won't work. There was a moment when that was true, I remember commenting as such, but I think it was that change that I reverted. I'll take another look.

Did it fail when you tested it?

**Comment by kjbrooks** [comment]
@jamienoss Yeah, it failed when I tested it. I think the error might still be on PyCharm if you want to take a look.

**Pull request merged**
